client/client.go

```diff
@@ -0,0 +1,106 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	websocket "github.com/gorilla/websocket"
```

Comment:
Haven't taken a close look, but is the Gorilla toolkit library preferable to https://godoc.org/golang.org/x/net/websocket?